grakn-factory/titan-factory/src/main/java/ai/grakn/graph/internal/GraknTitanGraph.java

```diff
@@ -90,8 +91,25 @@ protected void clearGraph() {
 
     @Override
     public void commitTransactionInternal(){
+        executeLockingMethod(super::commitTransactionInternal);
+    }
+
+    @Override
+    Vertex addVertex(Schema.BaseType baseType){
+        final Vertex[] v = new Vertex[1];
+        executeLockingMethod(() -> v[0] = super.addVertex(baseType));
+        return v[0];
```

Comment:
Both approaches seem pretty bad, so I'll approve this. If we end up using this method a lot more, we can reconsider.